File: legacy-python/app/explanations.py
def format_option_explanations(question: dict) -> str:
    """Build a readable breakdown covering every answer choice."""
    correct = question["correctAnswer"]
    sections: list[str] = []

    summary = (question.get("explanation") or "").strip()
    if summary:
        sections.append(summary)

    for opt in question.get("options") or []:
        label = opt.get("label", "")
        text = (opt.get("text") or "").strip()
        detail = (opt.get("explanation") or "").strip()
        if label == correct:
            header = f"{label}. {text} (Correct)"
            if not detail:
                detail = "This is the best answer."
        else:
            header = f"{label}. {text}"
            if not detail:
                detail = "This option is incorrect."
        sections.append(f"{header}\n{detail}")

    return "\n\n".join(sections)


def options_have_explanations(options: list[dict]) -> bool:
    if len(options) != 5:
        return False
    return all((opt.get("explanation") or "").strip() for opt in options)

File: src/domain/explanations.rs
use crate::models::Question;

pub fn options_have_explanations(options: &[crate::models::QuestionOption]) -> bool {
    options.len() == 5 && options.iter().all(|o| !o.explanation.trim().is_empty())
}

pub fn format_option_explanations(question: &Question) -> String {
    let mut sections = Vec::new();
    if !question.explanation.trim().is_empty() {
        sections.push(question.explanation.trim().to_string());
    }
    for opt in &question.options {
        let header = if opt.label == question.correct_answer {
            format!("{}. {} (Correct)", opt.label, opt.text)
        } else {
            format!("{}. {}", opt.label, opt.text)
        };
        let detail = if opt.explanation.trim().is_empty() {
            if opt.label == question.correct_answer {
                "This is the best answer.".to_string()
            } else {
                "This option is incorrect.".to_string()
            }
        } else {
            opt.explanation.trim().to_string()
        };
        sections.push(format!("{header}\n{detail}"));
    }
    sections.join("\n\n")
}